src/system/esp32/network.c

```diff
@@ -12,271 +12,236 @@
  *   ADLINK zenoh team, <zenoh@adlink-labs.tech>
  */
 
-#include <lwip/sockets.h>
-#include <errno.h>
 #include <netdb.h>
-#include <net/if.h>
-#include <unistd.h>
+
 #include "zenoh-pico/system/common.h"
 #include "zenoh-pico/utils/private/logging.h"
 
-/*------------------ Interfaces and sockets ------------------*/
-char *_zn_select_scout_iface()
+/*------------------ Endpoint ------------------*/
+void _zn_release_tcp_endpoint(void *arg)
 {
-    // TODO: scouting not supported yet
-    char *result = NULL;
-    return result;
+    struct addrinfo *self = (struct addrinfo*)arg;
+    freeaddrinfo(self);
 }
 
-struct sockaddr_in *_zn_make_socket_address(const char *addr, int port)
+void _zn_release_udp_endpoint(void *arg)
 {
-    struct sockaddr_in *saddr = (struct sockaddr_in *)malloc(sizeof(struct sockaddr_in));
-    memset(saddr, 0, sizeof(struct sockaddr_in));
-    saddr->sin_family = AF_INET;
-    saddr->sin_port = htons(port);
+    struct addrinfo *self = (struct addrinfo*)arg;
+    freeaddrinfo(self);
+}
 
-    if (inet_pton(AF_INET, addr, &(saddr->sin_addr)) <= 0)
-    {
-        free(saddr);
+/*------------------ TCP sockets ------------------*/
+void* _zn_create_tcp_endpoint(const char *s_addr, const char *port)
```

Comment:
`_zn_create_tcp_endpoint` => `_zn_create_endpoint_tcp`

Reply:
Agree and done in the next commit.
I also uniformed all the other function names to follow the same naming approach.